module/os/freebsd/zfs/zfs_dir.c

```diff
@@ -911,7 +912,7 @@ zfs_get_xattrdir(znode_t *zp, znode_t **xzpp, cred_t *cr, int flags)
 	va.va_mode = S_IFDIR | S_ISVTX | 0777;
 	zfs_fuid_map_ids(zp, cr, &va.va_uid, &va.va_gid);
 
-	error = zfs_make_xattrdir(zp, &va, xzpp, cr);
+	error = zfs_make_xattrdir(zp, &va, xzpp, cr, B_FALSE);
 
```

Comment:
Thanks, I will update this for consistency. The boolean is not being used for FreeBSD but was added to make the build compatible.

Reply:
Hmm. I see Linux calls zfs_zaccess(), while FreeBSD doesn't at all, so there is nothing really to skip.  So you may ignore that comment of mine.  But now I wonder why is this there on Linux?

Reply:
I checked and it was added during the initial commit while porting zfs on Linux https://github.com/openzfs/zfs/commit/34dc7c2f2#diff-5c59a2f8d51fe5cd3ff56b5d01f01a9c7cd1a28b32a61a8bbc136bbc91e18da9R821. Since Linux kernel checks the permissions within VFS stack already, I am also not very sure if it's something we really need it here or maybe it covers some edge case that I am not aware of.
@behlendorf.

Reply:
@behlendorf Do you think if ```zfs_zaccess()``` should be kept in ```zfs_make_xattrdir()``` for Linux. FreeBSD does not have it and I am not sure if it's really required for Linux also.

Reply:
@ixhamza it seems to me this shouldn't be required for Linux either, and the VFS will be performing the needed access checks.  If you can do some manual testing to make sure there's nothing obvious we're overlooking I think we can drop this.  It's been so long I don't recall why it was originally added with the original port.

Reply:
@behlendorf Thank you. I did some testing by setting/retrieving extended attributes and manipulating ACLs for different users and VFS already seems to be checking all the required permissions. I have also updated the PR.